src/es-features/task6.js

```diff
@@ -5,47 +5,44 @@
  * console.log(task6Old()); -> ['Max', 100, 'Admin', false, '1']
  */
 
-export function task6Old() {
-  function userModule() {
-    return {
-      name: 'Max',
-      value: 100,
-      role: { name: 'Admin' },
-      cases: [{ id: '1' }],
-    };
-  }
+// export function task6Old() {
+//   function userModule() {
+//     return {
+//       name: 'Max',
+//       value: 100,
+//       role: { name: 'Admin' },
+//       cases: [{ id: '1' }],
+//     };
+//   }
 
-  var tmp = userModule();
-  var name = tmp.name;
-  var value = tmp.value;
-  var role = tmp.role.name;
-  var isActive = tmp.isActive === undefined ? false : tmp.isActive;
-  var firstCaseId = tmp.cases[0].id;
+//   var tmp = userModule();
+//   var name = tmp.name;
+//   var value = tmp.value;
+//   var role = tmp.role.name;
+//   var isActive = tmp.isActive === undefined ? false : tmp.isActive;
+//   var firstCaseId = tmp.cases[0].id;
 
-  return [name, value, role, isActive, firstCaseId];
-}
+//   return [name, value, role, isActive, firstCaseId];
+// }
 
 // Напишите реализацию функции task6Old на ES6+ ниже этого комментария.
 // При желании, можете использовать стрелочную функцию, вместо обычной
 
 export const task6New = () => {
-  const userModule = () => {
-    return {
-      name: 'Max',
-      value: 100,
-      role: { name: 'Admin' },
-      cases: [{ id: '1' }],
-    };
-  };
+  const userModule = () => ({
+    name: 'Max',
+    value: 100,
+    role: { name: 'Admin' },
+    cases: [{ id: '1' }],
+  });
+
   const user = userModule();
   const {
     name,
     value,
     role: { name: occupation },
     cases: [{ id }],
   } = user;
-  const isActive = user.isActive === undefined ? false : user.isActive;
+  const isActive = user.isActive === true;
```

Comment:
still can be done more accurately. look closely to the original logic, what does it look like and what instruments we use for that? (you used it multiple time already in this homework)